### ISSUE TITLE
MacOS and Windows handle #:type-name differently

### ISSUE THREAD

**Issue by default-kramer** [racket/typed-racket#1142]
### What version of Racket are you using?
Racket CS 8.2

### What program did you run?
```
(struct replay ([frame : Frame]
                [old-frame : Frame]
                [action-queue : Queue])
  #:transparent
  #:mutable
  #:type-name Replay)

(: replay-enqueue! (-> replay (U (Stamped Action) (Listof (Stamped Action))) Void))
<... definition omitted ...>
```

### What should have happened?
On Windows the code compiles. On Mac I get the compile error `Type Checker: parse error in type; type name 'replay' is unbound`. (The error was easily fixed by using the type name `Replay` instead of the struct ID `replay`.)

I don't have a Mac available locally. I noticed the problem in Github Actions. My Windows workflow `runs-on: window-2019` and my Mac workflow `runs-on: macos11`. If needed, more details about `runs-on` are available here: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on

**Comment by capfredf** [comment]
The behavior on mac seems right to me. I don't have access to a Windows machine/vm right now, but I will get back to this later next week.

**Comment by default-kramer** [comment]
Interesting. On my local Windows machine, your code compiles and runs. My Racket version is 8.0.0.11--2021-03-15(c0cfd32/a) [cs]. So I will upgrade to 8.2 and see if it stops compiling. And if I can't reproduce locally, perhaps I'll set up a small Github repo that demonstrates the issue within Github Actions.

**Comment by default-kramer** [comment]
Oops, my bad. My Windows workflow actually did have the compile error but it got discarded and the workflow reported success. So the only version of Racket that has this bug was the snapshot build I was using locally.

**Comment by capfredf** [comment]
No problem! We fixed the bug in 8.1. see [the updated doc](https://docs.racket-lang.org/ts-reference/special-forms.html#%28part._.Structure_.Definitions%29) vs [the 8.0 doc](https://download.racket-lang.org/releases/8.0/doc/ts-reference/special-forms.html?q=struct#(form._((lib._typed-racket%2Fbase-env%2Fprims..rkt)._struct)))
